redis/src/main/scala/zio/redis/Input.scala

```diff
@@ -54,192 +62,220 @@ object Input {
+    def encode(data: (A, List[A]))(implicit codec: Codec): Chunk[RespValue.BulkString] =
       (data._1 :: data._2).foldLeft(Chunk.empty: Chunk[RespValue.BulkString])((acc, a) => acc ++ input.encode(a))
   }
 
   case object OrderInput extends Input[Order] {
-    def encode(data: Order): Chunk[RespValue.BulkString] = Chunk.single(encodeString(data.stringify))
+    def encode(data: Order)(implicit codec: Codec): Chunk[RespValue.BulkString] =
+      Chunk.single(encodeString(data.stringify))
   }
 
   case object RadiusUnitInput extends Input[RadiusUnit] {
-    def encode(data: RadiusUnit): Chunk[RespValue.BulkString] = Chunk.single(encodeString(data.stringify))
+    def encode(data: RadiusUnit)(implicit codec: Codec): Chunk[RespValue.BulkString] =
+      Chunk.single(encodeString(data.stringify))
   }
 
   case object RangeInput extends Input[Range] {
-    def encode(data: Range): Chunk[RespValue.BulkString] =
+    def encode(data: Range)(implicit codec: Codec): Chunk[RespValue.BulkString] =
       Chunk(encodeString(data.start.toString), encodeString(data.end.toString))
   }
 
   case object RegexInput extends Input[Regex] {
-    def encode(data: Regex): Chunk[RespValue.BulkString] = Chunk(encodeString("MATCH"), encodeString(data.regex))
+    def encode(data: Regex)(implicit codec: Codec): Chunk[RespValue.BulkString] =
+      Chunk(encodeString("MATCH"), encodeString(data.regex))
   }
 
   case object ReplaceInput extends Input[Replace] {
-    def encode(data: Replace): Chunk[RespValue.BulkString] = Chunk.single(encodeString(data.stringify))
+    def encode(data: Replace)(implicit codec: Codec): Chunk[RespValue.BulkString] =
+      Chunk.single(encodeString(data.stringify))
   }
 
   case object StoreDistInput extends Input[StoreDist] {
-    def encode(data: StoreDist): Chunk[RespValue.BulkString] = Chunk(encodeString("STOREDIST"), encodeString(data.key))
+    def encode(data: StoreDist)(implicit codec: Codec): Chunk[RespValue.BulkString] =
+      Chunk(encodeString("STOREDIST"), encodeString(data.key))
   }
 
   case object StoreInput extends Input[Store] {
-    def encode(data: Store): Chunk[RespValue.BulkString] = Chunk(encodeString("STORE"), encodeString(data.key))
+    def encode(data: Store)(implicit codec: Codec): Chunk[RespValue.BulkString] =
+      Chunk(encodeString("STORE"), encodeString(data.key))
   }
 
   case object ScoreRangeInput extends Input[ScoreRange] {
-    def encode(data: ScoreRange): Chunk[RespValue.BulkString] =
+    def encode(data: ScoreRange)(implicit codec: Codec): Chunk[RespValue.BulkString] =
       Chunk(encodeString(data.min.stringify), encodeString(data.max.stringify))
   }
 
   case object StringInput extends Input[String] {
-    def encode(data: String): Chunk[RespValue.BulkString] = Chunk.single(encodeString(data))
+    def encode(data: String)(implicit codec: Codec): Chunk[RespValue.BulkString] = Chunk.single(encodeString(data))
+  }
+
+  final case class ArbitraryInput[A: Schema]() extends Input[A] {
```

Comment:
You can even make it an "explicit implicit", e.g.

```scala
final case class ArbitraryInput[A](implicit schema: Schema[A]) extends Input[A] { ... }
```

That way you don't have to use the summoner.

Reply:
Done